pkg/reducer/interface.go

```diff
@@ -23,11 +23,25 @@ type IntervalWindow interface {
 	EndTime() time.Time
 }
 
+// CreateReducer is the interface which is used to create a Reducer.
+type CreateReducer interface {
+	// Create creates a Reducer, will be invoked once for every keyed window.
+	Create() Reducer
+}
+
 // Reducer is the interface of reduce function implementation.
 type Reducer interface {
 	Reduce(ctx context.Context, keys []string, reduceCh <-chan Datum, md Metadata) Messages
 }
 
+// CreateReducerFunc is a utility type used to convert a create function to a CreateReducer type.
+type CreateReducerFunc func() Reducer
+
+// Create implements the function of CreateReducer.
+func (rf CreateReducerFunc) Create() Reducer {
+	return rf()
+}
```

Comment:
```suggestion
// CreateReducerFn is a utility type used to convert a create function to a CreateReducer type.
type CreateReducerFn func() Reducer

// Create implements the function of CreateReducer.
func (rf CreateReducerFn) Create() Reducer {
	return rf()
}
```

xxxFn sounds better than xxxFunc